packages/babel-generator/src/generators/methods.ts

```diff
@@ -1,12 +1,20 @@
 import type Printer from "../printer";
-import { isIdentifier } from "@babel/types";
 import type * as t from "@babel/types";
+import { isIdentifier } from "@babel/types";
 
 export function _params(
   this: Printer,
   node: t.Function | t.TSDeclareMethod | t.TSDeclareFunction,
+  idNode: t.Node,
+  parentNode: t.Node,
```

Comment:
Can we use something like `NodePath<t.Function | t.TSDeclareMethod | t.TSDeclareFunction>["parent"]`? It would be great if the `idNode` gets stronger typings.